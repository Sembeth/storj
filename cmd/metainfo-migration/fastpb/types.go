// Copyright (C) 2021 Storj Labs, Inc.
// See LICENSE for copying information.

// Package fastpb contains generated marshaling for Pointers.
package fastpb

import (
	"storj.io/common/storj"
)

//go:generate protoc -I=. --gogofaster_out=Mgoogle/protobuf/any.proto=github.com/gogo/protobuf/types,Mgoogle/protobuf/duration.proto=github.com/gogo/protobuf/types,Mgoogle/protobuf/struct.proto=github.com/gogo/protobuf/types,Mgoogle/protobuf/timestamp.proto=github.com/gogo/protobuf/types,Mgoogle/protobuf/wrappers.proto=github.com/gogo/protobuf/types,paths=source_relative:. pointerdb.proto

// NodeID is a unique node identifier.
type NodeID = storj.NodeID

// PieceID is a unique identifier for pieces.
type PieceID = storj.PieceID
