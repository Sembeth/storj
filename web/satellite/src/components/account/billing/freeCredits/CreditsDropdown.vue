// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="credits-dropdown" v-click-outside="closeDropdown">
        <div @click="redirect" class="credits-dropdown__link-container">
            <span class="credits-dropdown__link-container__link">Credits History</span>
        </div>
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import { RouteConfig } from '@/router';

@Component
export default class CreditsDropdown extends Vue {
    /**
     * Holds logic to redirect user to credit history page.
     */
    public redirect(): void {
        this.$router.push(RouteConfig.Account.with(RouteConfig.CreditsHistory).path);
    }

    /**
     * Closes dropdown.
     */
    public closeDropdown(): void {
        this.$emit('close');
    }
}
</script>

<style scoped lang="scss">
    .credits-dropdown {
        z-index: 120;
        position: absolute;
        left: 0;
        top: 35px;
        background-color: #fff;
        border-radius: 6px;
        border: 1px solid #c5cbdb;
        box-shadow: 0 8px 34px rgba(161, 173, 185, 0.41);
        width: 220px;

        &__link-container {
            width: calc(100% - 30px);
            height: 50px;
            padding: 0 15px;
            display: flex;
            align-items: center;
            border-radius: 6px;

            &:hover {
                background-color: #f5f5f7;
            }

            &__link {
                font-size: 14px;
                line-height: 19px;
                color: #7e8b9c;
            }
        }
    }
</style>
