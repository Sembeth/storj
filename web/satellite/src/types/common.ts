// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

export enum SortDirection {
    ASCENDING = 1,
    DESCENDING,
}

export class PartneredSatellite {
    constructor(
        public name: string = '',
        public address: string = '',
    ) {}
}
