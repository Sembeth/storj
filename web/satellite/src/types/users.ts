// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

/**
 * Exposes all user-related functionality.
 */
export interface UsersApi {
    /**
     * Updates users full name and short name.
     *
     * @param user - contains information that should be updated
     * @throws Error
     */
    update(user: UpdatedUser): Promise<void>;
    /**
     * Fetch user.
     *
     * @returns User
     * @throws Error
     */
    get(): Promise<User>;
}

/**
 * User class holds info for User entity.
 */
export class User {
    public id: string;
    public fullName: string;
    public shortName: string;
    public email: string;
    public partner: string;
    public partnerId: string;
    public password: string;

    public constructor(id: string = '', fullName: string = '', shortName: string = '', email: string = '', partner: string = '', partnerId: string = '', password: string = '') {
        this.id = id;
        this.fullName = fullName;
        this.shortName = shortName;
        this.email = email;
        this.partner = partner;
        this.partnerId = partnerId;
        this.password = password;
    }

    public getFullName(): string {
        return this.shortName === '' ? this.fullName : this.shortName;
    }
}

/**
 * User class holds info for updating User.
 */
export class UpdatedUser {
    public fullName: string;
    public shortName: string;

    public constructor(fullName: string, shortName: string) {
        this.fullName = fullName;
        this.shortName = shortName;
    }

    public setFullName(value: string) {
        this.fullName = value.trim();
    }

    public setShortName(value: string) {
        this.shortName = value.trim();
    }

    public isValid(): boolean {
        return !!this.fullName;
    }
}
