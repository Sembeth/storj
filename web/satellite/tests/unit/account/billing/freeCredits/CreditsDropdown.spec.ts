// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import sinon from 'sinon';
import { VNode } from 'vue';
import { DirectiveBinding } from 'vue/types/options';

import CreditsDropdown from '@/components/account/billing/freeCredits/CreditsDropdown.vue';

import { createLocalVue, mount } from '@vue/test-utils';

const localVue = createLocalVue();

let clickOutsideEvent: EventListener;

localVue.directive('click-outside', {
    bind: function (el: HTMLElement, binding: DirectiveBinding, vnode: VNode) {
        clickOutsideEvent = function(event: Event): void {
            if (el === event.target) {
                return;
            }

            if (vnode.context) {
                vnode.context[binding.expression](event);
            }
        };

        document.body.addEventListener('click', clickOutsideEvent);
    },
    unbind: function(): void {
        document.body.removeEventListener('click', clickOutsideEvent);
    },
});

describe('CreditsDropdown', (): void => {
    it('renders correctly', (): void => {
        const wrapper = mount(CreditsDropdown, {
            localVue,
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('clicks work correctly', async (): Promise<void> => {
        const clickSpy = sinon.spy();
        const wrapper = mount(CreditsDropdown, {
            localVue,
            methods: {
                redirect: clickSpy,
            },
        });

        await wrapper.find('.credits-dropdown__link-container').trigger('click');

        expect(clickSpy.callCount).toBe(1);
    });
});
