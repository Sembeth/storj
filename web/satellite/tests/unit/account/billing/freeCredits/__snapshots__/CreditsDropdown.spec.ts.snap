// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreditsDropdown renders correctly 1`] = `
<div class="credits-dropdown">
  <div class="credits-dropdown__link-container"><span class="credits-dropdown__link-container__link">Credits History</span></div>
</div>
`;
